HomeWork2/main.py

```diff
@@ -0,0 +1,135 @@
+import requests
+from copy import deepcopy
+import csv
+from collections import Counter
+from datetime import datetime, timedelta
+
+
+class MovieData:
+    FIELDNAMES = ['Title', 'Score', 'Popularity', 'Last_day_in_cinema']
+    HEADERS = {
+        'accept': 'application/json',
+        'Authorization': 'Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIzMTI3NGFmYTRlNTUyMjRjYzRlN2Q0NmNlMTNkOTZjOSIsInN1YiI6IjVkNmZhMWZmNzdjMDFmMDAxMDU5NzQ4OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.lbpgyXlOXwrbY0mUmP-zQpNAMCw_h-oaudAJB6Cn5c8'
+    }
+    BASE_PART = 'https://api.themoviedb.org/3/'
+
+    def __init__(self, pages):
+        self.url = f'{self.BASE_PART}discover/movie'
+        self.genre_data = self.get_genre_table()
+        self.data = []
+        self.get_data_from_pages(pages)
+        self.genre_table = self.genre_table_with_ids()
+
+    def get_response(self, number_of_pages):
+        params = {
+            'include_adult': False,
+            'include_video': False,
+            'sort_by': 'popularity.desc',
+            'page': number_of_pages
+        }
+        return requests.get(self.url, headers=self.HEADERS, params=params)
+
+    def get_genre_table(self):
+        genre_url = f'{self.BASE_PART}genre/movie/list'
+        return (requests.get(genre_url, headers=self.HEADERS)).json()
+
+    def get_data_from_pages(self, num_of_pages):
+        for i in range(1, num_of_pages + 1):
+            response = self.get_response(i)
+            self.data.extend(response.json()['results'])
+
+    def get_multiple_pages(self):
+        return self.data
+
+    def get_most_popular_title(self):
+        return max((param['popularity'], param['title']) for param in self.data)[1]
```

Comment:
max(self.data, key=lambda movie: movie['popularity'])['title']
another possible way for your consideration